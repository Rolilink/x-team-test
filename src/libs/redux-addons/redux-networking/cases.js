import { createCases } from '../redux-cases';
import networkFactory from './factory';

const { setSuccess, setError, setFetching } = createCases({
  setSuccess: state => ({ ...state, network: networkFactory() }),
  setError: (state, { payload }) => ({ ...state, network: networkFactory({ error: payload }) }),
  setFetching: state => ({ ...state, network: networkFactory({ isFetching: true }) }),
});

export default {
  setSuccess,
  setError,
  setFetching,
};
