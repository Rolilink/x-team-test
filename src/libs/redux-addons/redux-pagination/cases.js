import { createCases } from '../redux-cases';
import paginationFactory from './factory';

const { setPage, setLimit, incrementPage, decrementPage } = createCases({
  setPage: (state, { payload }) => ({
    ...state,
    pagination: paginationFactory({ page: payload.pagination.page, limit: state.pagination.limit }),
  }),
  setLimit: (state, { payload }) => ({
    ...state,
    pagination: paginationFactory({ limit: payload.pagination.limit, page: state.pagination.page }),
  }),
  incrementPage: state => ({
    ...state,
    pagination: paginationFactory({ page: state.pagination.page + 1, limit: state.limit }),
  }),
  decrementPage: state => ({
    ...state,
    pagination: paginationFactory({
      page: state.pagination.page < 1 ? 0 : state.pagination.page - 1,
      limit: state.pagination.limit,
    }),
  }),
});

export default {
  setPage,
  setLimit,
  incrementPage,
  decrementPage,
};
