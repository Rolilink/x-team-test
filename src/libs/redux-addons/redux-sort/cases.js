import _ from 'lodash';
import sortingFactory from './factory';
import { createCases } from '../redux-cases';

const { clearSort, setField, unsetField } = createCases({
  clearSort: state => ({ ...state, sort: sortingFactory() }),
  setField: (state, { payload }) =>
    ({
      ...state,
      sort: sortingFactory({ ...state.sort, ...payload.sort }),
    }),
  unsetField: (state, { payload }) => ({
    ...state,
    sort: sortingFactory(_.omit(state.sort, payload.sort.key)),
  }),
});

export default {
  clearSort,
  setField,
  unsetField,
};
