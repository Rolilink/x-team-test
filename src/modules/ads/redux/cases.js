import { createCases } from '../../../libs/redux-addons/redux-cases';

const { addAd, clearAds } = createCases({
  addAd: (state, { payload }) => ({ ...state, ads: [...state.ads, payload.ad] }),
  clearAds: state => ({ ...state, ads: [] }),
});

export {
  addAd,
  clearAds,
};
