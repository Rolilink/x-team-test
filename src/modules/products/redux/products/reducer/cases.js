import { cases as networkCases } from '../../../../../libs/redux-addons/redux-networking';
import { cases as paginationCases } from '../../../../../libs/redux-addons/redux-pagination';
import { createCase } from '../../../../../libs/redux-addons/redux-cases';

export const addProductsToProductsById = (state, { payload }) => ({
  ...state,
  productsById: payload.products.reduce(
    (productsById, product) => ({
      ...productsById,
      [product.id]: product,
    }),
    state.productsById,
  ),
});

const onProductFetchSuccess = createCase([
  networkCases.setSuccess,
  addProductsToProductsById,
  paginationCases.setPage,
]);

const maybeOnFetchSuccess = (state, action) => {
  const stateSort = Object.keys(state.sort)[0];
  const actionSort = Object.keys(action.payload.sort)[0];

  if (stateSort !== actionSort) {
    return state;
  }

  return onProductFetchSuccess(state, action);
};

export const onFetchSuccess = maybeOnFetchSuccess;
