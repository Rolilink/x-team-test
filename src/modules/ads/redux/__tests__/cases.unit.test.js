import { addAd, clearAds } from '../cases';

const state = {
  ads: [1, 2],
};

describe('Ads.Redux.Cases.addAd', () => {
  it('should add an ad at the end of the ads array', () => {
    const action = { payload: { ad: 3 } };

    expect(addAd(state, action).ads).toMatchSnapshot();
  });
});

describe('Ads.Redux.Cases.clearAds', () => {
  it('should add an ad at the end of the ads array', () => {
    const action = {};

    expect(clearAds(state, action).ads).toMatchSnapshot();
  });
});
