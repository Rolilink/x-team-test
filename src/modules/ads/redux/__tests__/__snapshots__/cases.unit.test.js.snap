// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ads.Redux.Cases.addAd should add an ad at the end of the ads array 1`] = `
Array [
  1,
  2,
  3,
]
`;

exports[`Ads.Redux.Cases.clearAds should add an ad at the end of the ads array 1`] = `Array []`;
